### ISSUE TITLE
PHP 7 incompatibility: Fatal error: 'break' not in the 'loop' or 'switch' context

### ISSUE THREAD

**Issue by rwetzlmayr** [SocialEngine/phpv4-issues#409]
What are the steps to reproduce this issue?
-------------------------------------------
1. Install SE 4.8.10 on a webserver running PHP 7.
2. Navigate to the site's landing page.

What happens?
-------------
An error message appears:

> Fatal error: 'break' not in the 'loop' or 'switch' context in application/libraries/Engine/View/Helper/HtmlLink.php on line 79

What were you expecting to happen?
----------------------------------
Flawless operation on PHP 7 ;)

What versions of software are you using?
----------------------------------------
**SocialEngine PHP Version:** 4.8.10

Here's a patch to fix this issue:

```diff
--- a/application/libraries/Engine/View/Helper/HtmlLink.php
+++ b/application/libraries/Engine/View/Helper/HtmlLink.php
@@ -72,23 +72,17 @@
     }
     if( !empty($params['QUERY']) ) {
       if( is_string($params['QUERY']) ) {
-        $params['QUERY'] = trim($params['QUERY'], '?&');
+        $href .= '?' . trim($params['QUERY'], '?&');
       } else if( is_array($params['QUERY']) ) {
-        $params['QUERY'] = http_build_query($params['QUERY']);
-      } else {
-        break;
+        $href .= '?' . http_build_query($params['QUERY']);
       }
-      $href .= '?' . $params['QUERY'];
     }
     if( !empty($params['HASH']) ) {
       if( is_string($params['HASH']) ) {
-        $params['HASH'] = ltrim($params['HASH'], '#');
-      } else {
-        break;
+        $href .= '#' . ltrim($params['HASH'], '#');
       }
-      $href .= '#' . $params['HASH'];
     }
 
     return $href;
   }
-}
\ No newline at end of file
+}
```

**Comment by Anshuse** [comment]
Hi @rwetzlmayr , 

Thanks for providing a patch. We have forwarded this fix to our development team and they will look into it to add this in our core module.

Thanks!

**Comment by smazurov** [comment]
This has been addressed in [4.8.11](http://blog.socialengine.com/2016/05/26/socialengine-php-4-8-11-is-released/)
